### DISCUSSION TITLE
[Feature]: How to configure antd for on-demand loading

### DISCUSSION THREAD

**Discussion by guossw** [remix-run/remix#2756] [Q&A]
### What is the new or updated feature that you are suggesting?

yeah，in order to use antd，I need to do：

```ts
import antdStyles from 'antd/dist/antd.css'; 

export function links() {
  return [
    { rel: 'stylesheet', href: antdStyles },
  ];
}
```
This way will load the styles of all components。

[babel-plugin-import](https://github.com/umijs/babel-plugin-import) solve this problem，But I don't know how to configure it in remix。

ths

### Why should this feature be included?

use antd & babel-plugin-import

**Comment by carlassmann** [comment]
You cannot configure a babel plugin in remix, since remix does not use babel. It uses esbuild. 

There is a super alpha [esbuild-plugin](https://www.npmjs.com/package/esbuild-dynamic-import-plugin/v/0.0.12), but until this gets more stable, I would recommend importing the styles like you described and as Ant Design [recommends in their docs](https://ant.design/docs/react/introduce#Usage).

**Comment by quinn** [comment]
@ccssmnn I think the doc page you linked to has changed, because `antd/dist/antd.css` no longer exists in the `antd` package. It seems now they rely exclusively on some other mechanism for loading styles which injects style tags, resulting in a FOUC. The FOUC is only visible on reload, when the content is displayed before the js has finished loading.

I made an issue here: https://github.com/ant-design/ant-design/issues/40087

**Comment by quinn** [comment]
I got it to work like this (warning it is terrible):

```js
import { createCache, extractStyle, StyleProvider } from '@ant-design/cssinjs'
import { Segmented } from 'antd'

export async function loader({ request }: LoaderArgs) {
	const cache = createCache()

	renderToString(
		<StyleProvider cache={cache}>
			<Segmented options={[1, 2]} />
		</StyleProvider>
	)

	const styleKeys = Array.from(cache.cache.keys()).filter(function (key) {
		return key.startsWith('style%')
	})

	const styleText = styleKeys
		.map((key) => {
			return (cache.cache.get(key)?.[1] as string[])[0]
		})
		.join('\n')

	return json(
		{ styleText },
	)
}

export default function App() {
	const { styleText } = useLoaderData<typeof loader>()

	return (
		<html lang="en" className="h-full">
			<head>
				<Meta />
				<Links />
				<style dangerouslySetInnerHTML={{ __html: styleText }} />
			</head>
			...
```

**Reply by devbytyler** [reply]
Did you ever find a better solution to this? I'd really like to support dynamic themes in my remix app without the FOUC.

**Reply by quinn** [reply]
@devbytyler Yes, I stopped using react and now render everything on the server and use HTMX. I've never been happier
